src/main/java/pl/wrapper/parking/facade/domain/ParkingServiceImpl.java

```diff
@@ -1,16 +1,95 @@
-    //   dokumentacja + swagger do enpointa/endpointów i testy jednostkowe + test integracyjny dla waszego
-    // enpointa/enpoiontów
+
+    @Override
+    public Result<ParkingResponse> getByName(String name, Boolean opened) {
+        Optional<ParkingResponse> found = findParking(parking -> Objects.equals(name, parking.name()),opened);
+        log.info("All parkings has been checked for name");
+
+        if (found.isPresent()) return handleFoundParking(found.get());
+
+        log.info("No parking found for name: {}", name);
+        return Result.failure(new ParkingError.ParkingNotFoundByName(name));
+    }
+
+    @Override
+    public Result<ParkingResponse> getById(Integer id, Boolean opened) {
+        Optional<ParkingResponse> found = findParking(parking -> Objects.equals(id, parking.parkingId()),opened);
+        log.info("All parkings has been checked for id");
+
+        if (found.isPresent()) return handleFoundParking(found.get());
+
+        log.info("No parking found for id: {}", id);
+        return Result.failure(new ParkingError.ParkingNotFoundById(id));
+    }
+
+    @Override
+    public Result<ParkingResponse> getBySymbol(String symbol, Boolean opened) {
+        Optional<ParkingResponse> found = findParking(parking -> Objects.equals(symbol, parking.symbol()),opened);
+        log.info("All parkings has been checked for symbol");
+
+        if (found.isPresent()) return handleFoundParking(found.get());
+
+        log.info("No parking found for symbol: {}", symbol);
+        return Result.failure(new ParkingError.ParkingNotFoundBySymbol(symbol));
+    }
+
+    @Override
+    public Result<List<ParkingResponse>> getByParams(String symbol, Integer id, String name, Boolean opened) {
+        Predicate<ParkingResponse> predicate = generatePredicateForParams(symbol, id, name);
+
+        List<ParkingResponse> list = pwrApiServerCaller.fetchData().stream()
+                .filter(predicate)
+                .filter(parking -> handleIsOpened(opened, parking.isOpened()))
+                .toList();
+
+        log.info("All parkings has been checked for all params");
+        return Result.success(list);
+    }
+
+    private Optional<ParkingResponse> findParking(Predicate<ParkingResponse> predicate, Boolean opened){
+        return pwrApiServerCaller.fetchData().stream()
+                .filter(predicate)
+                .filter(parking -> handleIsOpened(opened, parking.isOpened()))
+                .findFirst();
+    }
+
+    private boolean handleIsOpened(Boolean opened, Boolean isParkingOpened){
+        if (opened == null) return true;
+        return opened == isParkingOpened;
+    }
+
+    private Result<ParkingResponse> handleFoundParking(ParkingResponse found){
+        log.info("Parking found");
+        return Result.success(found);
+    }
+
+    private Predicate<ParkingResponse> generatePredicateForParams(String symbol,Integer id ,String name){
```

Comment:
add isOpened to this predicate generator so we avoid iterating through the collection twice